File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.js
function solution(l, r) {
    let queue = [5];
    let result = [];
    
    while (queue.length > 0) {
        let num = queue.shift();
        
        if (num >= l && num <= r) {
            result.push(num);
        }
        
        let next1 = num * 10;
        let next2 = num * 10 + 5;
        
        if (next1 <= r) queue.push(next1);
        if (next2 <= r) queue.push(next2);
    }
    
    return result.length > 0 ? result : [-1];
}

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.*;

class Solution {
    public int[] solution(int l, int r) {
        List<Integer> result = new ArrayList<>();
        
        for (int i = l; i <= r; i++) {
            if (isValidate(i)) {
                result.add(i);
            }
        }
        
        if (result.isEmpty()) {
            return new int[]{-1};
        }
        
        return result.stream().mapToInt(Integer::intValue).toArray();
    }
    
    private boolean isValidate(int num) {
        while (num > 0) {
            int digit = num % 10;
            if (digit != 0 && digit != 5) {
                return false;
            }
            num /= 10;
        }
        return true;
    }
}